src/mongo/MongoCommands.ts

```diff
@@ -18,16 +18,61 @@ import { MongoFindOneResultEditor } from './editors/MongoFindOneResultEditor';
+		output.show();
+		output.appendLine("Running all commands in scrapbook...")
+		let commands = MongoCommands.getAllCommandsFromActiveEditor();
+		await MongoCommands.executeCommands(vscode.window.activeTextEditor, database, extensionPath, editorManager, tree, context, commands);
+	}
+
 	public static async executeCommandFromActiveEditor(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext): Promise<void> {
+		const commands = MongoCommands.getAllCommandsFromActiveEditor();
 		const activeEditor = vscode.window.activeTextEditor;
-		if (activeEditor.document.languageId !== 'mongo') {
-			vscode.window.showInformationMessage("This command can only be run inside of a MongoDB scrapbook (*.mongo)");
-			return;
-		}
 		const selection = activeEditor.selection;
-		const command = MongoCommands.getCommand(activeEditor.document.getText(), selection.start);
+		const command = MongoCommands.findCommandAtPosition(commands, selection.start);
+		return this.executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
+	}
+
+	public static async executeCommandFromText(database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext, commandText: string): Promise<void> {
+		const activeEditor = vscode.window.activeTextEditor;
+		const command = MongoCommands.getCommandFromText(commandText, new vscode.Position(0, 0));
+		return this.executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
+	}
+
+	public static getAllCommandsFromActiveEditor(): MongoCommand[] {
+		const activeEditor = vscode.window.activeTextEditor;
+		if (activeEditor) {
+			const commands = MongoCommands.getAllCommandsFromTextDocument(activeEditor.document);
+			return commands;
+		} else {
+			throw new Error(notInScrapbookMessage);
+		}
+	}
+
+	public static getAllCommandsFromTextDocument(document: vscode.TextDocument): MongoCommand[] {
+		if (document.languageId !== 'mongo') {
+			throw new Error(notInScrapbookMessage);
+		}
+
+		return MongoCommands.getAllCommands(document.getText());
+	}
+
+	public static async executeCommands(activeEditor: vscode.TextEditor, database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext, commands: MongoCommand[]): Promise<void> {
+		for (let command of commands) {
+			await this.executeCommand(activeEditor, database, extensionPath, editorManager, tree, context, command);
+		}
+	}
+
+	public static async executeCommand(activeEditor: vscode.TextEditor, database: IAzureParentNode<MongoDatabaseTreeItem>, extensionPath, editorManager: CosmosEditorManager, tree: AzureTreeDataProvider, context: IActionContext, command: MongoCommand): Promise<void> {
 		if (command) {
+			output.show();
```

Comment:
Not a fan of this output - did you add it for debugging purposes only? A few reasons:
1. The user should know what command we're running and it feels like a bug if we're not running the command they expect
1. There's no way to link this output to its corresponding result.json file
1. We're already popping up the result.json file and I think popping up the output channel as well is too much

I guess I'm okay with this as long as you remove `output.show`.

Another idea is to insert the command to the top of the result.json file as a comment (although comments aren't strictly supported in json files so idk)

Reply:
Personally I much prefer to see what it's doing, especially for Execute All, and if it hits an error.  Fine with not showing output, though.  We might need to refine this behavior in some way. We can revisit after people have played with it more.

Reply:
Good point - but it feels like we should include the command in the error message anyways

Reply:
I still think we need to play with it and make holistic changes in the flow rather than dealing with it in this PR.  The error handling is not new behavior, although execute all is new.

Reply:
That's fine - you already removed `output.show()` which was the "new" behavior that I cared about the most